### PR TITLE
#1023 - Fix all structural properties ignored when using `$expand` without `$select` and camel cased property names

### PR DESCRIPTION
**Pull request by orty**
This fixes the issue #1023 where all structural properties are removed from the generated SQL query when using `$expand` without `$select` in a `builder.EnableLowerCamelCase()` scenario.

Probably fixes the same issue when using aliases in an EdmModel with the `[DataMember(Name=...)]` attribute.

### CODE REVIEW

**Comment by ElizabethOkerio** [comment]
@orty Thanks for the contribution. Please add some tests.

File: src/Microsoft.AspNetCore.OData/Microsoft.AspNetCore.OData.xml [hunk 1]
```diff
@@ -13961,7 +13961,6 @@
         <member name="P:Microsoft.AspNetCore.OData.Routing.ODataRouteOptions.Order">
             <summary>
             Gets or sets the route order in conventional routing.
-            By default, move the conventional routing later as much as possible.
```
Comment (ElizabethOkerio) [inline src/Microsoft.AspNetCore.OData/Microsoft.AspNetCore.OData.xml:13964]: why this change?

Comment (orty) [reply src/Microsoft.AspNetCore.OData/Microsoft.AspNetCore.OData.xml:13964]: I just rebuilt the solution to validate my fix, and the comment must have been removed in a previous commit without committing the regeneration of the documentation XML. I can revert it if, but it might come back the next time someone rebuilds the solution.

**Comment by orty** [comment]
> @orty Thanks for the contribution. Please add some tests.

Done, I added a new lower cased model and all its context, plus a copy of your previous test method in this context, checking in both scenario that the top-level properties are not `null` or `default`

**Review by xuzhg — APPROVED** [review]

**Comment by ElizabethOkerio** [comment]
The build is not running. Let me see whether I can manually run it.
